hibernate-core/src/main/java/org/hibernate/generator/internal/SourceGeneration.java

```diff
@@ -74,7 +74,7 @@ public SourceGeneration(SourceType sourceType, Class<?> propertyType, GeneratorC
 	 */
 	@Override
 	public EnumSet<EventType> getEventTypes() {
-		return INSERT_ONLY;
+		return INSERT_AND_UPDATE;
```

Comment:
Oops, I didn't actually mean to push this fix to a bug reported in https://hibernate.atlassian.net/browse/HHH-17751 as part of this PR .... but it's done now so whatever.